### PR TITLE
Add gnu auto generated test cases to the generator

### PR DESCRIPTION
**Pull request by yulong-plct**
Hi,@eopXD:
This PR add the codes that can generate gnu test cases to the generator.
I did it together with @Incarnation-p-lee and @xl0714.
After that, I would also submit PR about creating folders, adding headers and checker.
Please review it. Thanks!

### CODE REVIEW

**Review by eopXD — COMMENTED** [review]
The CI currently does not pass for this single commit. I think a whole series of commit just like what you did in #272 would be good, with the comments addressed and commits reorganized.

**Comment by yulong-plct** [comment]
OK.I'll complete the other commits later.

**Comment by yulong-plct** [comment]
I added the other two commits about creating folders and adding headers.  We consider to create anther PR for adding checker, because we need to fully consider how to better interpret the magic number in the check information.@eopXD @Incarnation-p-lee @xl0714

File: rvv-intrinsic-generator/Makefile [hunk 1]
```diff
@@ -108,7 +108,7 @@ endef
 define gen_tests
 	rm -f $(1)/*.c
 	$(PY3) -m $(MAIN) --gen $(2) \
-		--out $(1) $(3) \
+		--out $(1) $(3) $(4)\
```
Comment (eopXD) [inline rvv-intrinsic-generator/Makefile:111]: No extra parameter is needed in this function. Your patch runs successfully without this edit.

File: rvv-intrinsic-generator/rvv_intrinsic_gen/main.py [hunk 2]
```diff
@@ -96,7 +98,7 @@ class GenTypes:
           "overloaded-docs", "non-overloaded-test", "overloaded-test",
           "non-overloaded-compatible-header", "overloaded-compatible-header"
       ])
-  parser.add_argument("--llvm", default=False, action="store_true")
+  parser.add_argument("--toolchain-type", default=ToolChainType.UNKNOWN)
```
Comment (eopXD) [inline rvv-intrinsic-generator/rvv_intrinsic_gen/main.py:101]: Need code to stop the generator if no `--toolchain-type` is specified.

**Review by eopXD — COMMENTED** [review]
The PR should be good after the remaining comments are addressed.

File: rvv-intrinsic-generator/rvv_intrinsic_gen/main.py (hunk 2 above)
Comment (yulong-plct) [reply rvv-intrinsic-generator/rvv_intrinsic_gen/main.py:101]: I tried  add the code in the generator.APITestGenerator.Funs function, but will affect the auto generated test cases at the auto-generated/API folder.  If we make sure that the toolchain-type specified in the Makefile is correct, can we not add the corresponding code?

Comment (yulong-plct) [reply rvv-intrinsic-generator/rvv_intrinsic_gen/main.py:101]: I tried  add the code in the generator.APITestGenerator.Funs function, but will affect the auto generated test cases at the auto-generated/API folder.  If we make sure that the toolchain-type specified in the Makefile is correct, can we not add the corresponding code?

Comment (eopXD) [reply rvv-intrinsic-generator/rvv_intrinsic_gen/main.py:101]: I mean, handle error under main.py to exit and emit error when `--toolchain-type` is not specified.

Comment (yulong-plct) [reply rvv-intrinsic-generator/rvv_intrinsic_gen/main.py:101]: OK! I got it. I will add code.

Comment (eopXD) [reply rvv-intrinsic-generator/rvv_intrinsic_gen/main.py:101]: Thanks for the reminder, we can still generate test case, like what we have right now in `api-testing/`, that don't need LLVM or GNU test headers. So I think no modifications is needed here.

**Review by eopXD — APPROVED** [review]

**Pull request merged**